controllers/workspace/devworkspace_controller.go

```diff
@@ -271,6 +274,11 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	workspace.Spec.Template = *flattenedWorkspace
 	reconcileStatus.setConditionTrue(conditions.DevWorkspaceResolved, "Resolved plugins and parents from DevWorkspace")
 
+	// Notify if error occured when resolving external DevWorkspaceOperatorConfig, as long as no other active warnings are present
+	if condition, ok := reconcileStatus.conditions[conditions.DevWorkspaceWarning]; !ok || condition.Status == corev1.ConditionFalse {
+		reconcileStatus.setCondition(conditions.DevWorkspaceWarning, configResolveStatus.conditions[conditions.DevWorkspaceWarning])
+	}
+
```

Comment:
This is somewhat awkward placement for this -- if we want it here to not lose e.g. a warning in flattening, then it should be even later in the function. What if a later step wants to add a warning?

See the note about initializing reconcileStatus earlier, that may be a cleaner option. Note that for now, `setConditionTrue` overwrites the condition if it's already present, so taking this approach would have the same functionality.

Reply:
> What if a later step wants to add a warning?

Good point :+1: 

> Note that for now, setConditionTrue overwrites the condition if it's already present, so taking this approach would have the same functionality.

I think the issue (with setting the warning condition earlier) then becomes that the config resolution warning will be suppressed by the [no-warning-case when flattening](https://github.com/devfile/devworkspace-operator/blob/main/controllers/workspace/devworkspace_controller.go#L269).

Perhaps the workspace conditions need to be modified a bit to allow for enabling/disabling different types of conditions with more granularity (eg. allow for a flattening warning condition set to false, while having a config resolution warning set to true)

Reply:
Ah that's a good point... I might err on the side of removing that line in the flatten flow and never setting the warning condition as false :shrug: 

Up to you. I've got a TODO on my backlog somewhere to improve warnings handling.

Reply:
Though it's not _nice_, I'm also +1 for removing that line from the flattening flow, for the time being.

I can create another issue to improve warnings handling, and I'm down to work on this whenever :)

Reply:
I didn't realize the warning will get unset if no warnings occurred by the end of the reconcile function - so this isn't as big of a deal as I had initially thought